docs/topics/multiplatform/multiplatform-run-tests.md

```diff
@@ -1,67 +1,375 @@
+> * It should not matter which framework you use to run tests in `commonTest`. However, running these tests with each
+>   framework you intend to use may help check that your development environment is correctly set up.
+> * When writing tests for platform-specific code, it's possible to use the annotations and extensions provided by a
+>   specific testing framework you want to use.
+>
+{type="tip"}
+
+## Deep dive into multiplatform testing
+
+### Writing tests for common code
+
+Consider `CurrentRuntime`, a type that holds the details of the platform on which the code is executed. For example, it
+might have the values "OpenJDK" and "17.0" for Android unit tests that run on a local JVM.
+
+An instance of `CurrentRuntime` should be created with the name and version of the platform as strings, where the
+version is optional. When the version is present, you only need the number at the start of the string, if available.
+
+1. In the `commonMain/kotlin` folder, create a new `org.kmp.testing` directory and the `CurrentRuntime.kt` file.
+2. Update it with the following implementation:
+
+    ```kotlin
+    class CurrentRuntime(val name: String, rawVersion: String?) {
+        companion object {
+            val versionRegex = Regex("^[0-9]+(\\.[0-9]+)?")
+        }
+    
+        val version = parseVersion(rawVersion)
+    
+        override fun toString() = "$name version $version"
+    
+        private fun parseVersion(rawVersion: String?): String {
+            val result = rawVersion?.let { versionRegex.find(it) }
+            return result?.value ?: "unknown"
+        }
+    }
+    ```
+
+3. In `commonTest`, create the new `org.kmp.testing/CurrentRuntimeTest.kt` file.
+4. Update it with a platform and framework-agnostic test like this:
 
-class SampleTestsIOS {
-    @Test
-    fun testHello() {
-        assertTrue("iOS" in hello())
+    ```kotlin
+    import kotlin.test.Test
+    import kotlin.test.assertEquals
+
+    class CurrentRuntimeTest {
+        @Test
+        fun shouldDisplayDetails() {
+            val runtime = CurrentRuntime("MyRuntime", "1.1")
+            assertEquals("MyRuntime version 1.1", runtime.toString())
+        }
+    
+        @Test
+        fun shouldHandleNullVersion() {
+            val runtime = CurrentRuntime("MyRuntime", null)
+            assertEquals("MyRuntime version unknown", runtime.toString())
+        }
+    
+        @Test
+        fun shouldParseNumberFromVersionString() {
+            val runtime = CurrentRuntime("MyRuntime", "1.2 Alpha Experimental")
+            assertEquals("MyRuntime version 1.2", runtime.toString())
+        }
+    
+        @Test
+        fun shouldHandleMissingVersion() {
+            val runtime = CurrentRuntime("MyRuntime", "Alpha Experimental")
+            assertEquals("MyRuntime version unknown", runtime.toString())
+        }
     }
-}
+    ```
+
+You can run this test using any of the ways [available in the IDE](#run-tests).
+
+### Writing platform-specific tests
+
```

Comment:
Maybe add a short introduction explaining that now the reader has covered two common code tests, it is time to create a platform-specific code test?